### PR TITLE
[PT FE] Added support for aten::renorm

### LINKED ISSUE #29692: [Good First Issue]: Support aten::embedding_renorm_
### Context

As in the main issue.

### What needs to be done?

As in the main issue.

### Example Pull Requests

_No response_

### Resources

- [Contribution guide - start here!](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Intel DevHub Discord channel](https://discord.gg/7pVRxUwdWG) - engage in discussions, ask questions and talk to OpenVINO developers
- [How to link your Pull Request to an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#manually-linking-a-pull-request-to-an-issue-using-the-pull-request-sidebar)

### Contact points

As in the main issue.

### Ticket

_No response_

### PR DESCRIPTION
**Pull request by Labman42**
### Description
This PR adds support for the aten::renorm operation in the PyTorch frontend of OpenVINO.

### Details:
 - *Implemented aten::renorm in renorm.cpp*
 - *add renorm unit test*

### Tickets:
 - Resolve #29692

### CODE REVIEW

**Comment by p-wysocki** [comment]
bump @mvafin this PR looks good at first glance. :)

File: tests/layer_tests/pytorch_tests/test_renorm.py
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_renorm.py:1]: Please add copyright here

File: src/frontends/pytorch/src/op/renorm.cpp
Comment (mvafin) [inline src/frontends/pytorch/src/op/renorm.cpp:1]: Please add copyright here

File: src/frontends/pytorch/src/op/renorm.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+namespace op {
+
+using namespace ov::op;
+
+OutputVector translate_renorm(const NodeContext& context) {
+    // aten::renorm(Tensor self, float p, int dim, float maxnorm) -> Tensor
+    num_inputs_check(context, 4, 4);
+    auto input = context.get_input(0);
+    auto p_val = get_input_with_floating_type(context, 1);
+    int32_t dim_val = context.const_input<int32_t>(2);
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/renorm.cpp:23]: No need to get din as constant. Use `get_input` and use as axis for `ReduceSum`

**Review by mvafin — CHANGES_REQUESTED** [review]
Code looks good. But it enables operation `aten::renorm`, not `aten::embedding_renorm_`. Please still finish the PR as this is still a useful change

File: src/frontends/pytorch/src/op/renorm.cpp
Comment (Labman42) [reply src/frontends/pytorch/src/op/renorm.cpp:1]: copyright added

File: src/frontends/pytorch/src/op/renorm.cpp (hunk 1 above)
Comment (Labman42) [reply src/frontends/pytorch/src/op/renorm.cpp]: changed to get_input

**Comment by Labman42** [comment]
Thanks for taking a look at the PR. I noticed that the issue specifically mentions `aten::embedding_renorm_`. I’m having trouble finding the proper reference or documentation for `aten::embedding_renorm_` in the PyTorch codebase. Could someone point me to the relevant reference or provide guidance on what is expected for this operation?

**Comment by mvafin** [comment]
> Thanks for taking a look at the PR. I noticed that the issue specifically mentions `aten::embedding_renorm_`. I’m having trouble finding the proper reference or documentation for `aten::embedding_renorm_` in the PyTorch codebase. Could someone point me to the relevant reference or provide guidance on what is expected for this operation?

To create this operation you can use `torch.ops.aten.embedding_renorm_`. But to understand what it does I can only suggest to study the pytorch code as I am not aware of any documentation for it.

File: src/frontends/pytorch/src/op/renorm.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    num_inputs_check(context, 4, 4);
+    auto input = context.get_input(0);
+    auto p_val = get_input_with_floating_type(context, 1);
+    auto dim_input = context.get_input(2);
+
+    auto maxnorm_val = get_input_with_floating_type(context, 3);
+    auto abs_input = context.mark_node(std::make_shared<v0::Abs>(input));
+    auto one_const = context.mark_node(v0::Constant::create(element::f32, Shape{}, {1.0f}));
+    auto one_i32 = context.mark_node(v0::Constant::create(element::i32, Shape{}, {1}));
+    auto axis = context.mark_node(std::make_shared<v0::Xor>(one_i32, dim_input));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/renorm.cpp:33]: Why do you xor here?

File: tests/layer_tests/pytorch_tests/test_renorm.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import pytest
+import torch
+import numpy as np
+
+from pytorch_layer_test_class import PytorchLayerTest
+
+class TestRenorm(PytorchLayerTest):
+    def _prepare_input(self, m, n):
+        return (np.random.randn(m, n).astype(np.float32),)
```
[the remark this answers is not in the document]
Comment (mvafin) [reply tests/layer_tests/pytorch_tests/test_renorm.py:12]: torch.renorm can accept tensors with more then 2 dimensions
